### PR TITLE
[#1] Create Slack API listener Module

### PR DESCRIPTION
**Pull request by tuha-2062**
### Working Issue (*)
Before: #1

### What i did (*)
- [x] Create Slack API listener Module

### [Checklist](https://github.com/the-cev7/checklist-code-review#heavy_check_mark-1-pull-request-has-been-self-reviewed) (*)
- [x] Pull request has been self-reviewed
- [x] Check impacted areas

### [UnitTest](https://github.com/the-cev7/checklist-code-review#heavy_check_mark-6-self-describing-test-method) (*)
- [x] Self-describing test method
- [x] My tests are fast!

### [Security](https://github.com/the-cev7/checklist-code-review#heavy_check_mark-12-validate-all-data-sent-from-the-client) (*)
- [x] Validate all data sent from the client
- [x] Output has no information about SQL query

### [Performance](https://github.com/the-cev7/checklist-code-review#heavy_check_mark-16-resolved-n--1-query)  (Optional)
- [x] Resolved n + 1 query
- [x] Time open page < 1000 ms

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -15,3 +15,4 @@ DB_DEBUG=true
 DOCKER_UID=USER_ID
 DOCKER_GID=GROUP_ID
 
+SLACK_SIGNING_SECRET=8f742231b10e8888abcd99yyyzzz85a5
```
Comment (dungda-0794) [inline .env.example]: cái này nên xoá đi em nhé

File: config/slack.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"go-app/pkg/logger"
+
+	"github.com/spf13/viper"
+)
+
+// SlackConfig to hold serect/key/config of slack
+type SlackConfig struct {
+	SigningSecret string `mapstructure:"SLACK_SIGNING_SECRET"`
```
Comment (dungda-0794) [inline config/slack.go:11]: Good

File: internal/domain/slack.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package domain
+
+import (
+	"context"
+)
+
+// SlackAppMentionEvent entity
+// https://api.slack.com/events/app_mention
+type SlackAppMentionEvent struct {
+	Type    string `json:"type"`
```
Comment (dungda-0794) [inline internal/domain/slack.go:10]: Good

File: .env.example (hunk 1 above)
Comment (tuha-2062) [reply .env.example]: @dungda-0794 https://api.slack.com/authentication/verifying-requests-from-slack
cái này key fake em lấy ở sample của họ thôi, giữ cũng được đúng không anh

File: internal/modules/slack/delivery/http/middleware.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+	"github.com/labstack/echo/v4"
+)
+
+// TimeNow have to export this var for testing purpose
+var TimeNow = time.Now
+
+// SlackVerifyingRequests is Middleware verifying request from slack
+func SlackVerifyingRequests(config config.SlackConfig) echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
```
Comment (dungda-0794) [inline internal/modules/slack/delivery/http/middleware.go:24]: Good

**Review by dungda-0794 — APPROVED** [review]
LGTM

**Pull request merged**